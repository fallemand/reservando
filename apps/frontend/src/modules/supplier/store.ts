/**
 * This file provides the store module to handle
 * the user status between different vue apps
 * */
/* eslint-disable @typescript-eslint/explicit-function-return-type, @typescript-eslint/no-explicit-any */
import { ActionTree, GetterTree, MutationTree, Module } from "vuex";
import { GlobalType } from "@Types/globals";

declare const GLOBAL: GlobalType;

export interface SupplierStatusState {
  isLoggedIn: boolean;
  isStaff: boolean;
  privileges: string[];
  supplierId: number;
  isOnline: boolean;
  isManaged: boolean;
  isOutbound: boolean;
  isConfirmed: boolean;
  priority: number;
  supplierName: string;
  canAddProducts: boolean;
  hiddenInformationBoxes: { [key: string]: string[] };
  hasCompanyRegistration?: boolean;
  hasBusinessInsurance?: boolean;
  hasPaymentInfo?: boolean;
  hasCompanyLogo?: boolean;
  containerName: string;
  anonymizedIp: string;
  countryCode: string;
  visitorId: string;
  sessionId: string;
}

const state: SupplierStatusState = {
  isLoggedIn: false,
  isStaff: false,
  privileges: [],
  supplierId: 0,
  isOnline: true,
  isManaged: false,
  isOutbound: false,
  isConfirmed: false,
  priority: null,
  supplierName: "",
  canAddProducts: false,
  hiddenInformationBoxes: {},
  hasCompanyRegistration: false,
  hasBusinessInsurance: true,
  hasPaymentInfo: false,
  hasCompanyLogo: true,
  containerName: "",
  anonymizedIp: "",
  countryCode: null,
  visitorId: "",
  sessionId: "",
};

const getters: GetterTree<SupplierStatusState, any> = {
  isBoxVisible: (state) => (page: string, boxId: string) => {
    if (!Object.keys(state.hiddenInformationBoxes).length) {
      return true;
    }
    return !(
      state.hiddenInformationBoxes[`supplier:${page}`] &&
      state.hiddenInformationBoxes[`supplier:${page}`].includes(boxId)
    );
  },
  hasPrivilege: (state) => (privilege: string) =>
    state.privileges.includes(privilege) || state.isStaff,
};

const actions: ActionTree<SupplierStatusState, any> = {
  setSupplierStatus({ commit }, status) {
    commit("SET_SUPPLIER_STATUS", status);
  },
};

const mutations: MutationTree<SupplierStatusState> = {
  SET_SUPPLIER_STATUS(state, status) {
    Object.assign(state, status);
  },
  SET_HIDDEN_INFORMATION_BOXES(state, hiddenInformationBoxes) {
    state.hiddenInformationBoxes = hiddenInformationBoxes;
  },
};

const module: Module<SupplierStatusState, any> = {
  namespaced: true,
  state,
  actions,
  getters,
  mutations,
};

export default module;
