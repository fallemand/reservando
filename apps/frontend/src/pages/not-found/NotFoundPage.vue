<template>
  <Page v-bind="pageProps">
    <div class="page-not-found__graphic page-not-found__graphic--webp" />
    <h3 class="page-not-found__title">
      {{ $t("pNotFound_cText_BodyTitle") }}
    </h3>
    <p class="page-not-found__text">
      {{ $t("pNotFound_cText_BodyText") }}
    </p>
  </Page>
</template>

<script lang="ts" src="./not-found.ts" />

<style lang="scss">
@import "~@Assets/styles/variables";
@import "~@Assets/styles/mixins";

@import "./not-found";
</style>
