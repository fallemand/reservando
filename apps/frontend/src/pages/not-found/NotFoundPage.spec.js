import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import supplier from "@Modules/supplier/store";
import NotFoundPage from "./NotFoundPage";

const localVue = createLocalVue();
localVue.use(Vuex);

describe("NotFoundPage", () => {
  let storeData;

  beforeEach(() => {
    storeData = {
      modules: {
        supplier,
      },
    };
  });

  it("should render", () => {
    const wrapper = shallowMount(NotFoundPage, { store: new Vuex.Store(storeData), localVue });
    expect(cleanSnapshot(wrapper)).toMatchSnapshot();
  });
});
