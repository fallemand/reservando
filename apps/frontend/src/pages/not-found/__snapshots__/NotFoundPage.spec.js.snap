// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage should render 1`] = `
"<page-stub title='' pagetitle='pNotFound_cText_BodyTitle' id='not-found' type='normal' header='[object Object]' footer='[object Object]' tracking='[object Object]' infoboxvisible='[function cleaned]' hideinfobox='() => '$storeAction'' snackbarmessage='' snackbartype='error' snackbardelay='5000' hastitle='true'>
  <div class='page-not-found__graphic page-not-found__graphic--webp'></div>
  <h3 class='page-not-found__title'>
    pNotFound_cText_BodyTitle
  </h3>
  <p class='page-not-found__text'>
    pNotFound_cText_BodyText
  </p>
  <gygbutton-stub size='medium' modifier='outline-light' type='button' href='/' class='page-not-found__cta'>
    pNotFound_cText_BodyCta
  </gygbutton-stub>
</page-stub>
"
`;
